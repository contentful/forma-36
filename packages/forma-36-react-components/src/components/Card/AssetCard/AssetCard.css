@import 'settings/colors';
@import 'settings/typography';
@import 'settings/transitions';
@import 'settings/shadows';

.AssetCard {
  height: calc(1rem * (300 / var(--font-base-default)));
  display: inline-flex;
  padding: 0px;
  min-width: calc(1rem * (240 / var(--font-base-default)));
  transition: box-shadow var(--transition-duration-short)
      var(--transition-easing-default),
    border-color var(--transition-duration-default)
      var(--transition-easing-default);
}

.AssetCard--size-small {
  height: calc(
    1rem * (188 / var(--font-base-default))
  ); /* 188 is equal to our 142px small variant plus the height of the card header */
  min-width: calc(1rem * (142 / var(--font-base-default)));
}

.AssetCard--size-small .AssetCard__asset {
  height: calc(1rem * (142 / var(--font-base-default)));
  width: calc(1rem * (142 / var(--font-base-default)));
}

.AssetCard--drag-active {
  box-shadow: var(--box-shadow-heavy);
}

.AssetCard__asset {
  height: calc(
    1rem * (254 / var(--font-base-default))
  ); /* 254 Height of the container substracted by the border width of the header */
}

.AssetCard__wrapper {
  flex: 1 1 auto;
}

.AssetCard__header {
  padding: 0 calc(1rem * (14 / var(--font-base-default)));
  height: calc(1rem * (45 / var(--font-base-default)));
  display: flex;
  justify-content: flex-end;
  align-items: center;
  border-bottom: calc(1rem * (1 / var(--font-base-default))) solid
    var(--color-element-mid);
}

.AssetCard__status {
  margin-left: .5rem;
}

.AssetCard__actions {
  margin-left: calc(1rem * (10 / var(--font-base-default)));
}
