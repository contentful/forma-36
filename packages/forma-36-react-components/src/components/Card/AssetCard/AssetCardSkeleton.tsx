import React from 'react';

import {
  SkeletonBodyText,
  SkeletonContainer,
  SkeletonImage,
} from '../../Skeleton';

export interface AssetCardSkeletonProps {
  size?: 'small' | 'default';
}

export const AssetCardSkeleton = (props: AssetCardSkeletonProps) => (
  <SkeletonContainer svgWidth={props.size === 'small' ? 150 : 240}>
    <SkeletonImage
      offsetLeft={props.size === 'small' ? 40 : 85}
      offsetTop={props.size === 'small' ? 50 : 100}
    />
    <SkeletonBodyText
      offsetLeft={props.size === 'small' ? 25 : 70}
      offsetTop={props.size === 'small' ? 140 : 190}
      numberOfLines={1}
      width={100}
    />
  </SkeletonContainer>
);
