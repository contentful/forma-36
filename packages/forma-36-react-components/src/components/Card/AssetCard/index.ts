export * from './AssetCard';
export { default } from './AssetCard';
