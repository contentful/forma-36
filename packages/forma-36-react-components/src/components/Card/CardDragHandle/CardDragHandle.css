@import 'settings/colors';
@import 'settings/transitions';
@import 'settings/typography';
@import 'settings/borders';

.CardDragHandle {
  display: flex;
  position: relative;
  width: calc(1rem * (20 / var(--font-base-default)));
  height: 100%;
  align-items: center;
  background-color: var(--color-element-lightest);
  border: 0;
  padding: 0;
  border-right: 1px solid var(--color-element-mid);
  cursor: grab;
  border-top-left-radius: var(--border-radius-medium);
  border-bottom-left-radius: var(--border-radius-medium);
  transition: background-color var(--transition-duration-default)
    var(--transition-easing-default);
}

.CardDragHandle:hover,
.CardDragHandle:focus {
  background-color: var(--color-element-light);
}

.CardDragHandle--drag-active {
  background-color: var(--color-element-light);
  cursor: grabbing;
}

.CardDragHandle__sr-label {
  composes: sr-only from './../../../styles/settings/helpers.css';
}
