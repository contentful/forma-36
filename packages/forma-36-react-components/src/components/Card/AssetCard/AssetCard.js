import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import Card from '../Card';
import Dropdown from '../../Dropdown/Dropdown';
import Asset from '../../Asset';
import Icon from '../../Icon/Icon';
import TabFocusTrap from '../../TabFocusTrap';
import Tag from '../../Tag';
import AssetCardSkeleton from './AssetCardSkeleton';
import styles from './AssetCard.css';

class AssetCard extends React.Component {
  static propTypes = {
    extraClassNames: PropTypes.string,
    src: PropTypes.string,
    isLoading: PropTypes.bool,
    type: PropTypes.string,
    title: PropTypes.string,
    dropdownListElements: PropTypes.node,
    status: PropTypes.oneOf(['archived', 'changed', 'draft', 'published']),
    testId: PropTypes.string,
  };

  static defaultProps = {
    extraClassNames: undefined,
    dropdownListElements: undefined,
    isLoading: false,
    src: undefined,
    title: undefined,
    status: undefined,
    type: undefined,
    testId: 'cf-ui-asset-card',
  };

  state = {
    isOpen: false,
  };

  renderDropdownListElements = dropdownListElements => (
    <Dropdown
      isOpen={this.state.isOpen}
      onClose={() => this.setState({ isOpen: false })}
      position="bottom-right"
      toggleElement={
        <button
          type="button"
          className={styles['AssetCard__header__dropdown-button']}
          onClick={() => this.setState(state => ({ isOpen: !state.isOpen }))}
        >
          <TabFocusTrap>
            <Icon icon="MoreHorizontalTrimmed" color="secondary" />
          </TabFocusTrap>
        </button>
      }
    >
      {dropdownListElements}
    </Dropdown>
  );

  renderStatus = status => {
    let label;
    let type;

    switch (status) {
      case 'archived':
        label = 'archived';
        type = 'negative';
        break;

      case 'changed':
        label = 'changed';
        type = 'primary';
        break;

      case 'published':
        label = 'published';
        type = 'positive';
        break;

      default:
        label = 'draft';
        type = 'warning';
    }

    return (
      <Tag tagType={type} style={{ marginLeft: 'auto' }}>
        {label}
      </Tag>
    );
  };

  render() {
    const {
      extraClassNames,
      src,
      type,
      title,
      status,
      isLoading,
      dropdownListElements,
      testId,
      ...otherProps
    } = this.props;

    const classNames = cn(styles.AssetCard, extraClassNames);
    return (
      <Card
        extraClassNames={classNames}
        title={title}
        testId={testId}
        {...otherProps}
      >
        {isLoading ? (
          <AssetCardSkeleton />
        ) : (
          <div className={styles['AssetCard__inner-wrapper']}>
            <div className={styles.AssetCard__header}>
              {status && this.renderStatus(status)}
              {dropdownListElements &&
                this.renderDropdownListElements(dropdownListElements)}
            </div>
            <div className={styles.AssetCard__content}>
              <Asset
                extraClassNames={styles.AssetCard__asset}
                src={src}
                title={title}
                type={type}
              />
            </div>
          </div>
        )}
      </Card>
    );
  }
}

export default AssetCard;
