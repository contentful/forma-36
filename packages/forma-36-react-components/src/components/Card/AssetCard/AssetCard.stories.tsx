import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, select, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import AssetCard from './AssetCard';
import Icon from '../../Icon';
import DropdownList from '../../Dropdown/DropdownList';
import DropdownListItem from '../../Dropdown/DropdownListItem';
import CardDragHandle from './../CardDragHandle';
import { types } from '../../Asset/Asset';

storiesOf('Components/Card/AssetCard', module)
  .addParameters({
    propTypes: AssetCard['__docgenInfo'],
    component: AssetCard,
  })
  .add('default', () => (
    <AssetCard
      className={text('className', '')}
      status={select(
        'status',
        {
          Draft: 'draft',
          Changed: 'changed',
          Published: 'published',
          Archived: 'archived',
        },
        'published',
      )}
      type={select('type', types, 'image')}
      isLoading={boolean('isLoading', false)}
      src={text('src', 'https://placekitten.com/200/300')}
      title={text('title', 'Image of a cat')}
      withDragHandle={boolean('withDragHandle', false)}
      isDragActive={boolean('isDragActive', false)}
      size={select(
        'size',
        {
          small: 'small',
          default: 'default',
        },
        'default',
      )}
    />
  ))
  .add('with custom CardDragHandle', () => (
    <AssetCard
      className={text('className', '')}
      status={select(
        'status',
        {
          Draft: 'draft',
          Changed: 'changed',
          Published: 'published',
          Archived: 'archived',
        },
        'published',
      )}
      type={select('type', types, 'image')}
      isLoading={boolean('isLoading', false)}
      src={text('src', 'https://placekitten.com/200/300')}
      title={text('title', 'Image of a cat')}
      cardDragHandleComponent={<CardDragHandle>Reorder card</CardDragHandle>}
      isDragActive={boolean('isDragActive', false)}
      size={select(
        'size',
        {
          small: 'small',
          default: 'default',
        },
        'default',
      )}
    />
  ))
  .add('with dropdownListElements', () => (
    <AssetCard
      className={text('className', '')}
      status={select(
        'status',
        {
          Draft: 'draft',
          Changed: 'changed',
          Published: 'published',
          Archived: 'archived',
        },
        'published',
      )}
      type={select('type', types, 'image')}
      isLoading={boolean('isLoading', false)}
      src={text('src', 'https://placekitten.com/200/300')}
      title={text('title', 'Image of a cat')}
      withDragHandle={boolean('withDragHandle', false)}
      isDragActive={boolean('isDragActive', false)}
      dropdownListElements={
        <React.Fragment>
          <DropdownList>
            <DropdownListItem isTitle>Actions</DropdownListItem>
            <DropdownListItem onClick={action('Edit onClick')}>
              Edit
            </DropdownListItem>
            <DropdownListItem onClick={action('Download onClick')}>
              Download
            </DropdownListItem>
            <DropdownListItem onClick={action('Remove onClick')}>
              Remove
            </DropdownListItem>
          </DropdownList>
          <DropdownList border="top">
            <DropdownListItem isTitle>File info</DropdownListItem>
            <DropdownListItem>Asset name</DropdownListItem>
            <DropdownListItem>Asset type</DropdownListItem>
            <DropdownListItem>Asset size</DropdownListItem>
            <DropdownListItem>Asset resolution</DropdownListItem>
          </DropdownList>
        </React.Fragment>
      }
      size={select(
        'size',
        {
          small: 'small',
          default: 'default',
        },
        'default',
      )}
    />
  ))
  .add('with statusIcon', () => (
    <AssetCard
      className={text('className', '')}
      status={select(
        'status',
        {
          Draft: 'draft',
          Changed: 'changed',
          Published: 'published',
          Archived: 'archived',
        },
        'published',
      )}
      statusIcon="Clock"
      type={select('type', types, 'image')}
      isLoading={boolean('isLoading', false)}
      src={text('src', 'https://placekitten.com/200/300')}
      title={text('title', 'Image of a cat')}
      withDragHandle={boolean('withDragHandle', false)}
      isDragActive={boolean('isDragActive', false)}
      size={select(
        'size',
        {
          small: 'small',
          default: 'default',
        },
        'default',
      )}
    />
  ))
  .add('with a custom statusIcon', () => (
    <AssetCard
      className={text('className', '')}
      status={select(
        'status',
        {
          Draft: 'draft',
          Changed: 'changed',
          Published: 'published',
          Archived: 'archived',
        },
        'published',
      )}
      statusIcon={<Icon icon="Calendar" />}
      type={select('type', types, 'image')}
      isLoading={boolean('isLoading', false)}
      src={text('src', 'https://placekitten.com/200/300')}
      title={text('title', 'Image of a cat')}
      withDragHandle={boolean('withDragHandle', false)}
      isDragActive={boolean('isDragActive', false)}
      size={select(
        'size',
        {
          small: 'small',
          default: 'default',
        },
        'default',
      )}
    />
  ));
