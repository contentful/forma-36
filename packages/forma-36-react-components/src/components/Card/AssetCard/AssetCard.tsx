import React, { Component } from 'react';
import cn from 'classnames';
import Card, { BaseCardProps } from '../Card';
import Icon, { IconType } from '../../Icon';
import CardActions from './../CardActions';
import { Asset } from '../../Asset';
import { AssetType } from '../../Asset';
import Tag, { TagType } from '../../Tag';
import AssetCardSkeleton from './AssetCardSkeleton';
import CardDragHandle, { CardDragHandlePropTypes } from '../CardDragHandle';
import styles from './AssetCard.css';

export type AssetState = 'archived' | 'changed' | 'draft' | 'published';

export interface AssetCardProps extends BaseCardProps {
  /**
   * The source of the asset (will also render a thumbnail if the AssetCard's type is set to image)
   */
  src: string;
  /**
   * The title of the asset
   */
  title: string;
  /**
   * Loading state for the AssetCard - when true will display loading feedback to the user
   */
  isLoading?: boolean;
  /**
   * The DropdownList elements used to render an actions dropdown for the AssetCard
   */
  dropdownListElements?: React.ReactElement;
  /**
   * The publish status of the asset
   */
  status?: AssetState;
  /**
   * An icon for the status of the entry
   */
  statusIcon?: React.ReactNode;
  /**
   * The type of asset being represented
   */
  type?: AssetType;
  /**
   * Prop to pass a custom CardDragHandle component to for use in drag and drop contexts
   */
  cardDragHandleComponent?: React.ReactNode;
  /**
   * Renders a default drag handle for the component for use in drag and drop contexts
   */
  withDragHandle?: boolean;
  /**
   * Props to pass down to the default CardDragHandle component (does not work with cardDragHandleComponent prop)
   */
  cardDragHandleProps?: Partial<CardDragHandlePropTypes>;
  /**
   * Applies styling for when the component is actively being dragged by the user
   */
  isDragActive?: boolean;
  /**
   * Renders a small variant of the card which accommodates a 150x150px image
   */
  size?: 'small' | 'default';
}

const defaultProps: Partial<AssetCardProps> = {
  isLoading: false,
  testId: 'cf-ui-asset-card',
  size: 'default',
};

export class AssetCard extends Component<AssetCardProps> {
  static defaultProps = defaultProps;

  renderStatus = (status: AssetState, statusIcon: React.ReactNode) => {
    let label;
    let type: TagType | null = null;

    switch (status) {
      case 'archived':
        label = 'archived';
        type = 'negative';
        break;

      case 'changed':
        label = 'changed';
        type = 'primary';
        break;

      case 'published':
        label = 'published';
        type = 'positive';
        break;

      default:
        label = 'draft';
        type = 'warning';
    }

    return (
      <>
        {statusIcon && typeof statusIcon === 'string' ? (
          <Icon
            icon={statusIcon as IconType}
            color="muted"
            className="f36-margin-right--xs"
          />
        ) : (
          statusIcon
        )}
        <Tag className={styles['AssetCard__status']} tagType={type}>
          {label}
        </Tag>
      </>
    );
  };

  renderCardDragHandle() {
    const {
      cardDragHandleComponent,
      isDragActive,
      cardDragHandleProps,
      withDragHandle,
    } = this.props;

    if (cardDragHandleComponent) {
      return cardDragHandleComponent;
    } else if (withDragHandle) {
      return (
        <CardDragHandle isDragActive={isDragActive} {...cardDragHandleProps}>
          Reorder entry
        </CardDragHandle>
      );
    }
  }

  render() {
    const {
      className,
      src,
      type,
      title,
      status,
      statusIcon,
      isLoading,
      dropdownListElements,
      isDragActive,
      size,
      cardDragHandleProps,
      cardDragHandleComponent,
      withDragHandle,
      ...otherProps
    } = this.props;

    const classNames = cn(
      styles.AssetCard,
      {
        [styles['AssetCard--drag-active']]: isDragActive,
        [styles[`AssetCard--size-${size}`]]: size,
      },
      className,
    );

    return (
      <Card className={classNames} padding="none" title={title} {...otherProps}>
        {isLoading ? (
          <AssetCardSkeleton size={size} />
        ) : (
          <React.Fragment>
            {this.renderCardDragHandle()}
            <div className={styles['AssetCard__wrapper']}>
              <div className={styles['AssetCard__header']}>
                {status && this.renderStatus(status, statusIcon)}
                {dropdownListElements && (
                  <CardActions
                    className={styles['AssetCard__actions']}
                    iconButtonProps={{
                      onClick: (e) => e.stopPropagation,
                    }}
                  >
                    {dropdownListElements}
                  </CardActions>
                )}
              </div>
              <div className={styles['AssetCard__content']}>
                <Asset
                  className={styles['AssetCard__asset']}
                  src={src}
                  title={title}
                  type={type}
                  status={status}
                />
              </div>
            </div>
          </React.Fragment>
        )}
      </Card>
    );
  }
}

export default AssetCard;
