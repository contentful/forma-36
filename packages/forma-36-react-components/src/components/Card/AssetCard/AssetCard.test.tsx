import React from 'react';
import { render } from '@testing-library/react';
import { Calendar } from '@contentful/f36-icons';

import { axe } from '../../../utils/axeHelper';
import { AssetCard } from './AssetCard';
import { DropdownList, DropdownListItem } from '../../Dropdown';
import { CardDragHandle } from '../CardDragHandle/CardDragHandle';

it('renders the component', () => {
  const { container } = render(
    <AssetCard
      title="picture of a cat"
      src="https://via.placeholder.com/200x300"
    >
      AssetCard
    </AssetCard>,
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders the component with an additional class name', () => {
  const { container } = render(
    <AssetCard
      className="my-extra-class"
      src="https://via.placeholder.com/200x300"
      title="picture of a cat"
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('renders the component in loading state', () => {
  const { container } = render(
    <AssetCard
      isLoading
      src="https://via.placeholder.com/200x300"
      title="picture of a cat"
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('renders the component with status', () => {
  const { container } = render(
    <AssetCard
      status="archived"
      src="https://via.placeholder.com/200x300"
      title="picture of a cat"
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('renders the component with actions', () => {
  const { container } = render(
    <AssetCard
      dropdownListElements={
        <DropdownList>
          <DropdownListItem isTitle>Actions</DropdownListItem>
          <DropdownListItem onClick={() => {}}>Edit</DropdownListItem>
        </DropdownList>
      }
      src="https://via.placeholder.com/200x300"
      title="picture of a cat"
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('renders the component without actions', () => {
  const { container } = render(
    <AssetCard
      src="https://via.placeholder.com/200x300"
      title="picture of a cat"
    />,
  );
  const dropdown = container.querySelector('[data-test-id="cf-ui-dropdown"]');

  expect(dropdown).not.toBeTruthy();
  expect(container.firstChild).toMatchSnapshot();
});

it('renders the component with a drag handle', () => {
  const { container } = render(
    <AssetCard
      className="my-extra-class"
      src="https://via.placeholder.com/200x300"
      title="picture of a cat"
      withDragHandle
    />,
  );

  expect(container).toMatchSnapshot();
});

it('renders the component with a custom drag handle', () => {
  const { container } = render(
    <AssetCard
      className="my-extra-class"
      src="https://via.placeholder.com/200x300"
      title="picture of a cat"
      cardDragHandleComponent={<CardDragHandle>Reorder card</CardDragHandle>}
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('renders a small variant of the component', () => {
  const { container } = render(
    <AssetCard
      size="small"
      src="https://via.placeholder.com/200x300"
      title="picture of a cat"
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('has no a11y issues', async () => {
  const { container } = render(
    <AssetCard
      src="https://via.placeholder.com/200x300"
      title="picture of a cat"
    />,
  );
  const results = await axe(container);

  expect(results).toHaveNoViolations();
});

it('renders a component with a status icon', async () => {
  const { container } = render(
    <AssetCard
      src="https://via.placeholder.com/200x300"
      title="picture of a cat"
      statusIcon={Calendar}
    />,
  );
  const results = await axe(container);

  expect(results).toHaveNoViolations();
});
