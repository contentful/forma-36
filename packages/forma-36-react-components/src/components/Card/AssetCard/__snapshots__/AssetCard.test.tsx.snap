// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a small variant of the component 1`] = `
<div
  aria-label="picture of a cat"
  class="Card AssetCard AssetCard--size-small Card--padding-none"
  data-test-id="cf-ui-asset-card"
  title="picture of a cat"
>
  <div
    class="AssetCard__wrapper"
  >
    <div
      class="AssetCard__header"
    />
    <div
      class="AssetCard__content"
    >
      <div
        class="Asset AssetCard__asset"
        data-test-id="cf-ui-asset"
      >
        <div
          class="Asset__image-container"
        >
          <img
            alt="picture of a cat"
            class="Asset__image-container__image"
            src="http://placekitten.com/200/300"
          />
        </div>
        <div
          class="Asset__title-container"
        >
          <span
            class="Asset__title-container__title"
          >
            picture of a cat
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the component 1`] = `
<div
  aria-label="picture of a cat"
  class="Card AssetCard AssetCard--size-default Card--padding-none"
  data-test-id="cf-ui-asset-card"
  title="picture of a cat"
>
  <div
    class="AssetCard__wrapper"
  >
    <div
      class="AssetCard__header"
    />
    <div
      class="AssetCard__content"
    >
      <div
        class="Asset AssetCard__asset"
        data-test-id="cf-ui-asset"
      >
        <div
          class="Asset__image-container"
        >
          <img
            alt="picture of a cat"
            class="Asset__image-container__image"
            src="http://placekitten.com/200/300"
          />
        </div>
        <div
          class="Asset__title-container"
        >
          <span
            class="Asset__title-container__title"
          >
            picture of a cat
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the component in loading state 1`] = `
<div
  aria-label="picture of a cat"
  class="Card AssetCard AssetCard--size-default Card--padding-none"
  data-test-id="cf-ui-asset-card"
  title="picture of a cat"
>
  <svg
    aria-label="Loading component..."
    class="SkeletonContainer"
    data-test-id="cf-ui-skeleton-form"
    height="100%"
    preserveAspectRatio="xMidYMid meet"
    role="img"
    width="240"
  >
    <title>
      Loading component...
    </title>
    <rect
      clip-path="url(#f36-asset-card-skeleton)"
      height="100%"
      style="fill: url(#cf-ui-skeleton-clip-gradient-0);"
      width="100%"
      x="0"
      y="0"
    />
    <defs>
      <clippath
        id="f36-asset-card-skeleton"
      >
        <rect
          height="70"
          rx="0"
          ry="0"
          width="70"
          x="85"
          y="100"
        />
        <rect
          height="16"
          rx="0"
          ry="0"
          width="100"
          x="70"
          y="190"
        />
      </clippath>
      <lineargradient
        id="cf-ui-skeleton-clip-gradient-0"
      >
        <stop
          offset="0%"
          stop-color="#e5ebed"
          stop-opacity="1"
        >
          <animate
            attributeName="stop-color"
            dur="2s"
            id="animation-1"
            repeatCount="indefinite"
            values="#e5ebed; #f7f9fa; #e5ebed"
          />
        </stop>
        <stop
          offset="50%"
          stop-color="#f7f9fa"
          stop-opacity="1"
        >
          <animate
            attributeName="stop-color"
            begin="animation-1.begin+0.25s"
            dur="2s"
            repeatCount="indefinite"
            values="#e5ebed; #f7f9fa; #e5ebed"
          />
        </stop>
        <stop
          offset="100%"
          stop-color="#e5ebed"
          stop-opacity="1"
        >
          <animate
            attributeName="stop-color"
            begin="animation-1.begin+0.5s"
            dur="2s"
            repeatCount="indefinite"
            values="#e5ebed; #f7f9fa; #e5ebed"
          />
        </stop>
      </lineargradient>
    </defs>
  </svg>
</div>
`;

exports[`renders the component with a custom drag handle 1`] = `
<div
  aria-label="picture of a cat"
  class="Card AssetCard AssetCard--size-default my-extra-class Card--padding-none"
  data-test-id="cf-ui-asset-card"
  title="picture of a cat"
>
  <div
    class="CardDragHandle"
    data-test-id="cf-ui-card-drag-handle"
  >
    <svg
      class="Icon Icon--small Icon--muted"
      data-test-id="cf-ui-icon"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
    <span
      class="CardDragHandle__sr-label"
    >
      Reorder card
    </span>
  </div>
  <div
    class="AssetCard__wrapper"
  >
    <div
      class="AssetCard__header"
    />
    <div
      class="AssetCard__content"
    >
      <div
        class="Asset AssetCard__asset"
        data-test-id="cf-ui-asset"
      >
        <div
          class="Asset__image-container"
        >
          <img
            alt="picture of a cat"
            class="Asset__image-container__image"
            src="http://placekitten.com/200/300"
          />
        </div>
        <div
          class="Asset__title-container"
        >
          <span
            class="Asset__title-container__title"
          >
            picture of a cat
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the component with a drag handle 1`] = `
<div>
  <div
    aria-label="picture of a cat"
    class="Card AssetCard AssetCard--size-default my-extra-class Card--padding-none"
    data-test-id="cf-ui-asset-card"
    title="picture of a cat"
  >
    <div
      class="CardDragHandle"
      data-test-id="cf-ui-card-drag-handle"
    >
      <svg
        class="Icon Icon--small Icon--muted"
        data-test-id="cf-ui-icon"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
      <span
        class="CardDragHandle__sr-label"
      >
        Reorder entry
      </span>
    </div>
    <div
      class="AssetCard__wrapper"
    >
      <div
        class="AssetCard__header"
      />
      <div
        class="AssetCard__content"
      >
        <div
          class="Asset AssetCard__asset"
          data-test-id="cf-ui-asset"
        >
          <div
            class="Asset__image-container"
          >
            <img
              alt="picture of a cat"
              class="Asset__image-container__image"
              src="http://placekitten.com/200/300"
            />
          </div>
          <div
            class="Asset__title-container"
          >
            <span
              class="Asset__title-container__title"
            >
              picture of a cat
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the component with actions 1`] = `
<div
  aria-label="picture of a cat"
  class="Card AssetCard AssetCard--size-default Card--padding-none"
  data-test-id="cf-ui-asset-card"
  title="picture of a cat"
>
  <div
    class="AssetCard__wrapper"
  >
    <div
      class="AssetCard__header"
    >
      <div
        class="Dropdown AssetCard__actions"
        data-test-id="cf-ui-card-actions"
      >
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          class="IconButton IconButton--secondary"
          data-test-id="cf-ui-icon-button"
          type="button"
        >
          <span
            class="TabFocusTrap IconButton__inner"
            tabindex="-1"
          >
            <svg
              class="Icon Icon--small Icon--primary IconButton__icon"
              data-test-id="cf-ui-icon"
              height="18"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
            <span
              class="IconButton__label"
            >
              Actions
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      class="AssetCard__content"
    >
      <div
        class="Asset AssetCard__asset"
        data-test-id="cf-ui-asset"
      >
        <div
          class="Asset__image-container"
        >
          <img
            alt="picture of a cat"
            class="Asset__image-container__image"
            src="http://placekitten.com/200/300"
          />
        </div>
        <div
          class="Asset__title-container"
        >
          <span
            class="Asset__title-container__title"
          >
            picture of a cat
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the component with an additional class name 1`] = `
<div
  aria-label="picture of a cat"
  class="Card AssetCard AssetCard--size-default my-extra-class Card--padding-none"
  data-test-id="cf-ui-asset-card"
  title="picture of a cat"
>
  <div
    class="AssetCard__wrapper"
  >
    <div
      class="AssetCard__header"
    />
    <div
      class="AssetCard__content"
    >
      <div
        class="Asset AssetCard__asset"
        data-test-id="cf-ui-asset"
      >
        <div
          class="Asset__image-container"
        >
          <img
            alt="picture of a cat"
            class="Asset__image-container__image"
            src="http://placekitten.com/200/300"
          />
        </div>
        <div
          class="Asset__title-container"
        >
          <span
            class="Asset__title-container__title"
          >
            picture of a cat
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the component with status 1`] = `
<div
  aria-label="picture of a cat"
  class="Card AssetCard AssetCard--size-default Card--padding-none"
  data-test-id="cf-ui-asset-card"
  title="picture of a cat"
>
  <div
    class="AssetCard__wrapper"
  >
    <div
      class="AssetCard__header"
    >
      <div
        class="Tag Tag--negative"
        data-test-id="cf-ui-tag"
        style="margin-left: auto;"
      >
        archived
      </div>
    </div>
    <div
      class="AssetCard__content"
    >
      <div
        class="Asset AssetCard__asset"
        data-test-id="cf-ui-asset"
      >
        <div
          class="Asset__asset-container"
        >
          <div
            class="Asset__illustration-container"
          >
            <svg
              class="Illustration"
              data-test-id="cf-ui-asset-icon"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                stroke="#000"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-miterlimit="10"
              >
                <path
                  d="M19.5 23.5H.5V.5h13l6 6z"
                />
                <path
                  d="M13.5.5v6h6M13 11l-3 5.5-2-2-2.5 5h10z"
                />
                <circle
                  cx="7"
                  cy="9"
                  r="1.5"
                />
              </g>
            </svg>
          </div>
          <span
            class="Asset__asset-container__title"
          >
            picture of a cat
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the component without actions 1`] = `
<div
  aria-label="picture of a cat"
  class="Card AssetCard AssetCard--size-default Card--padding-none"
  data-test-id="cf-ui-asset-card"
  title="picture of a cat"
>
  <div
    class="AssetCard__wrapper"
  >
    <div
      class="AssetCard__header"
    />
    <div
      class="AssetCard__content"
    >
      <div
        class="Asset AssetCard__asset"
        data-test-id="cf-ui-asset"
      >
        <div
          class="Asset__image-container"
        >
          <img
            alt="picture of a cat"
            class="Asset__image-container__image"
            src="http://placekitten.com/200/300"
          />
        </div>
        <div
          class="Asset__title-container"
        >
          <span
            class="Asset__title-container__title"
          >
            picture of a cat
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
